core/src/main/java/io/questdb/cairo/PartitionBy.java

```diff
@@ -55,6 +77,150 @@ public static int fromString(CharSequence name) {
         return nameToIndexMap.get(name);
     }
 
+    public static PartitionAddMethod getPartitionAddMethod(int partitionBy) {
+        switch (partitionBy) {
+            case DAY:
+                return ADD_DD;
+            case MONTH:
+                return ADD_MM;
+            case YEAR:
+                return ADD_YYYY;
+            case HOUR:
+                return ADD_HH;
+            default:
+                return null;
+        }
+    }
+
+    public static DateFormat getPartitionDirFormatMethod(int partitionBy) {
+        switch (partitionBy) {
+            case DAY:
+                return fmtDay;
+            case MONTH:
+                return fmtMonth;
+            case YEAR:
+                return fmtYear;
+            case HOUR:
+                return fmtHour;
+            case NONE:
+                return fmtDefault;
+            default:
+                throw new UnsupportedOperationException("partition by " + partitionBy + " does not have date format");
+        }
+    }
+
+    public static PartitionFloorMethod getPartitionFloorMethod(int partitionBy) {
+        switch (partitionBy) {
+            case DAY:
+                return FLOOR_DD;
+            case MONTH:
+                return FLOOR_MM;
+            case YEAR:
+                return FLOOR_YYYY;
+            case HOUR:
+                return FLOOR_HH;
+            default:
+                return null;
+        }
+    }
+
+    public static boolean isPartitioned(int partitionBy) {
+        return partitionBy != NONE;
+    }
+
+    public static long parsePartitionDirName(CharSequence partitionName, int partitionBy) {
+        try {
+            return getPartitionDirFormatMethod(partitionBy).parse(partitionName, null);
+        } catch (NumericException e) {
+            final CairoException ee = CairoException.instance(0);
+            switch (partitionBy) {
+                case DAY:
+                    ee.put("'YYYY-MM-DD'");
+                    break;
+                case MONTH:
+                    ee.put("'YYYY-MM'");
+                    break;
+                case YEAR:
+                    ee.put("'YYYY'");
+                    break;
+                case HOUR:
+                    ee.put("'YYYY-MM-DDTHH'");
+                    break;
+            }
+            ee.put(" expected");
+            throw ee;
+        }
+    }
+
+    public static long setSinkForPartition(CharSink path, int partitionBy, long timestamp, boolean calculatePartitionMax) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 57 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.